src/librustc_typeck/check/wfcheck.rs

```diff
@@ -511,9 +535,8 @@ fn check_where_clauses<'a, 'gcx, 'fcx, 'tcx>(
                     .iter()
                     .flat_map(|p| ty::wf::predicate_obligations(fcx,
                                                                 fcx.param_env,
-                                                                fcx.body_id,
-                                                                p,
-                                                                span));
+                                                                &cause,
```

Comment:
This is fine, but `fcx.normalize_associated_types_in(span, &predicates)` above should be `fcx.inh.normalize_associated_types_in(cause, fcx.param_env, &predicates)` instead.